nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ValidateCsv.java

```diff
@@ -0,0 +1,433 @@
+                return new Equals();
+
+            case "forbidsubstr":
+                String[] forbiddenSubStrings = argument.split(",");
+                for (int i = 0; i < forbiddenSubStrings.length; i++) {
+                    forbiddenSubStrings[i] = forbiddenSubStrings[i].substring(1, forbiddenSubStrings[i].length() - 1);
+                }
+                return new ForbidSubStr(forbiddenSubStrings);
+
+            case "requiresubstr":
+                String[] requiredSubStrings = argument.split(",");
+                for (int i = 0; i < requiredSubStrings.length; i++) {
+                    requiredSubStrings[i] = requiredSubStrings[i].substring(1, requiredSubStrings[i].length() - 1);
+                }
+                return new RequireSubStr(requiredSubStrings);
+
+            case "strnotnullorempty":
+                if(argument != null && !argument.isEmpty())
+                    throw new IllegalArgumentException("StrNotNullOrEmpty does not expect any argument but has " + argument);
+                return new StrNotNullOrEmpty();
+
+            case "requirehashcode":
+                String[] hashs = argument.split(",");
+                int[] hashcodes = new int[hashs.length];
+                for(int i = 0; i < hashs.length; i++) {
+                    hashcodes[i] = Integer.parseInt(hashs[i]);
+                }
+                return new RequireHashCode(hashcodes);
+
+            case "null":
+                if(argument != null && !argument.isEmpty())
+                    throw new IllegalArgumentException("Null does not expect any argument but has " + argument);
+                return null;
+
+            default:
+                throw new IllegalArgumentException(method + " is not an allowed method to define a Cell Processor");
+        }
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) {
+        FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
+        }
+
+        final CsvPreference csvPref = this.preference.get();
+        final boolean header = context.getProperty(HEADER).asBoolean();
+        final ComponentLog logger = getLogger();
+        final CellProcessor[] cellProcs = this.processors.get();
+
+        final AtomicReference<Boolean> valid = new AtomicReference<Boolean>(true);
+
+        session.read(flowFile, new InputStreamCallback() {
+            @Override
+            public void process(final InputStream in) throws IOException {
+                CsvListReader listReader = null;
+                try {
+                    listReader = new CsvListReader(new InputStreamReader(in), csvPref);
+                    if(header) {
+                        listReader.read();
+                    }
+                    while(listReader.read(cellProcs) != null) {}
+                } catch (final IOException | SuperCsvCellProcessorException e) {
```

Comment:
This should break IOException and SuperCsvCellProcessorException into separate catch blocks. A SuperCsvCellProcessorException should be logged at DEBUG because it will get thrown for invalid files as a normal part of processing for this processor. 

For example, I used your template with the schema "Null, ParseDate("dd/MM/yyyy"), Optional(ParseDouble())". I had the second column have a value of "22/111954". It failed to parse the data and correctly routed it to invalid but logged an ERROR bulletin.